src/algebra/covariant_and_contravariant.lean

```diff
@@ -97,9 +95,166 @@ for all `m ∈ M` and all elements `n₁, n₂ ∈ N`, if the relation `r` holds
 `(μ m n₁, μ m n₂)` obtained from `(n₁, n₂)` by "acting upon it by `m`"", then, the relation `r`
 also holds for the pair `(n₁, n₂)`.
 
-If `m : M` and `h : r (μ m n₁) (μ m n₂)`, then `covariant_class.covc m h : r n₁ n₂`.
+If `m : M` and `h : r (μ m n₁) (μ m n₂)`, then `contravariant_class.covtc m h : r n₁ n₂`.
 -/
-class contravariant_class :=
-(covtc : contravariant M N μ s)
+class contravariant_class : Prop :=
+(covtc : contravariant M N μ r)
```

Comment:
Since I'm remaking about naming, what do `covtc` and `covc` stand for? maybe rename them `is_contravariant` or `contravariant` (with `@[protect_proj]`) or if you want something short you could even do `elim` (with `@[protect_proj]`)

Reply:
They are simply shortenings of `contravariant_class` and `covariant_class`: I wanted the keep the names short.  I will use `elim` instead.